metrics/storage/webtooling.sh

```diff
@@ -0,0 +1,194 @@
+#!/bin/bash
+#
+# Copyright (c) 2020-2021 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Description of the test:
+# This test runs the 'web tooling benchmark'
+# https://github.com/v8/web-tooling-benchmark
+
+#set -e
+set -x
+
+# General env
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_PATH}/../lib/common.bash"
+
+TEST_NAME="web-tooling"
+IMAGE="docker.io/library/local-web-tooling:latest"
+DOCKERFILE="${SCRIPT_PATH}/web-tooling-dockerfile/Dockerfile"
+CI_JOB="${CI_JOB:-""}"
+configuration_file="/usr/share/defaults/kata-containers/configuration.toml"
+# Directory to run the test on
+# This is run inside of the container
+TESTDIR="${TESTDIR:-/testdir}"
+file_path="/web-tooling-benchmark"
+file_name="output"
+CMD="mkdir -p ${TESTDIR}; cd $file_path && node dist/cli.js > $file_name"
+PAYLOAD_ARGS="${PAYLOAD_ARGS:-tail -f /dev/null}"
+NUM_CONTAINERS="$1"
+# This timeout is related with the amount of time that
+# webtool benchmark needs to run inside the container
+timeout=$((180 * "$NUM_CONTAINERS"))
+declare -a CONTAINERS_ID
+cpu_period="100000"
+cpu_quota="200000"
+
+TMP_DIR=$(mktemp --tmpdir -d webtool.XXXXXXXXXX)
+
+remove_tmp_dir() {
+	rm -rf "$TMP_DIR"
+}
+
+trap remove_tmp_dir EXIT
+
+# Show help about this script
+help(){
+cat << EOF
+Usage: $0 <count>
+   Description:
+        <count> : Number of containers to run.
+EOF
+}
+
+save_config(){
+	metrics_json_start_array
+
+	local json="$(cat << EOF
+	{
+		"containers": "$NUM_CONTAINERS",
+		"image": "$IMAGE",
+		"units": "runs/s"
+	}
+EOF
+)"
+	metrics_json_add_array_element "$json"
+	metrics_json_end_array "Config"
+}
+
+function main() {
+	# Verify enough arguments
+	if [ $# != 1 ]; then
+		echo >&2 "error: Not enough arguments [$@]"
+		help
+		exit 1
+	fi
+
+	containers=()
+	# Check tools/commands dependencies
+	cmds=("docker")
```

Comment:
This is correct, but it looks like we might be expecting it to change in the future? If so, it would be safer to do this "long hand":

```bash
cmds=()
cmds+=("docker")
```